app/src/main/java/com/davidtomas/taskyapp/features/agenda/presentation/agenda/UiAgendaItem.kt

```diff
@@ -0,0 +1,8 @@
+package com.davidtomas.taskyapp.features.agenda.presentation.agenda
+
+import com.davidtomas.taskyapp.features.agenda.domain.model.AgendaModel
+
+data class UiAgendaItem(
+    val agendaModel: AgendaModel,
+    val isDropdownOptionsShown: Boolean = false
+)
```

Comment:
To make this a bit simpler, you could also just add one `agendaItemWithOpenedOptions: AgendaItem? = null` to your root state. If it's not null, you know which agenda item to show the options for and if it's null you don't show any.
That way you also avoid accidentally showing it for multiple items which could happen with your approach